src/sensors/src/PredictSensorsMeasurements.cpp

```diff
@@ -0,0 +1,185 @@
+    // It is automatically proper acceleration since gravity is incorporated into the base acceleration
+    for(idx = 0; idx<numAccl; idx++)
+    {
+        accelerometer = (Accelerometer *)sensorsList.getSensor(iDynTree::ACCELEROMETER, idx);
+        int parentLinkId = accelerometer->getParentIndex();
+        
+        iDynTree::LinAcceleration measuredAcc, crossProdTerm;
+        iDynTree::Transform link_T_sensor;
+        iDynTree::SpatialAcc linkAccTerm;
+        iDynTree::Twist localVelocity;
+        returnVal = accelerometer->getLinkSensorTransform(link_T_sensor);
+        localVelocity = link_T_sensor*linkVel(parentLinkId);
+        linkAccTerm = link_T_sensor*(linkAcc(parentLinkId));
+        crossProdTerm = (localVelocity.getAngularVec3()).cross(localVelocity.getLinearVec3());
+        measuredAcc = (linkAccTerm).getLinearVec3() + crossProdTerm;       
+        predictedMeasurement.setMeasurement(iDynTree::ACCELEROMETER,idx,measuredAcc);
+        
+    }
+    for(idx = 0; idx<numGyro; idx++)
+    {
+        iDynTree::AngVelocity measuredAngVel;
+        gyroscope = (Gyroscope*)sensorsList.getSensor(iDynTree::GYROSCOPE, idx);
+        int parentLinkId = accelerometer->getParentIndex();
+        iDynTree::Transform link_T_sensor;
+        returnVal = gyroscope->getLinkSensorTransform(link_T_sensor);
+        measuredAngVel = (link_T_sensor*linkVel(parentLinkId)).getAngularVec3();
+        predictedMeasurement.setMeasurement(iDynTree::GYROSCOPE,idx,measuredAngVel);   
+        
+    }
+    return(returnVal);
+}
+
+bool PredictSensorsMeasurements::operator()(const Model& model,const Traversal& traversal,const iDynTree::FreeFloatingPos& robotPos,const iDynTree::FreeFloatingVel& robotVel,iDynTree::FreeFloatingAcc& robotAcc,const LinAcceleration& gravity,const iDynTree::SensorsList &sensorsList,iDynTree::VectorDynSize &predictedMeasurement)
+{
+    bool returnVal = true;
+    // incorporating gravity into the base LinAcceleration
+    iDynTree::LinkPositions linkPos(model);
+    iDynTree::LinkVelArray linkVel(model);
+    iDynTree::LinkAccArray linkAcc(model);
+    
+    iDynTree::AngAcceleration nullAngAccl;
+    iDynTree::SpatialAcc gravityAccl(gravity,nullAngAccl);
+    
+    robotAcc.baseAcc() = robotAcc.baseAcc() - gravityAccl;
+    
+    // calling ForwardPosVelAccKinematics with arguments
+    ForwardPosVelAccKinematics(model,traversal,robotPos,robotVel,robotAcc,linkPos,linkVel,linkAcc);
+    
+    // looping though SensorList    
+    unsigned int numAccl = sensorsList.getNrOfSensors(iDynTree::ACCELEROMETER);
+    unsigned int numGyro = sensorsList.getNrOfSensors(iDynTree::GYROSCOPE);
+    int idx;
+    
+    //Iterate through each kind of accelrometer and find its parent. Compute local (classical accelration) 
+    // It is automatically proper acceleration since gravity is incorporated into the base acceleration
+
+    
+    std::map<unsigned int,Sensor*> acceleroMap;
+    std::map<unsigned int,Sensor*> gyroMap;
```

Comment:
Actually all this method needed to be removed because it is just (untested) code duplication of the other `makePrediction` method, as discussed in the last point in https://github.com/robotology/idyntree/pull/104#issuecomment-163581506 . If you are ok with it I can work on this.